avm/res/kusto/cluster/main.bicep

```diff
@@ -325,8 +325,8 @@ resource kustoCluster_roleAssignments 'Microsoft.Authorization/roleAssignments@2
 ]
 
 module kustoCluster_principalAssignments 'principal-assignment/main.bicep' = [
-  for (principalAssignment, index) in (principalAssignments ?? []): {
-    name: '${uniqueString(deployment().name, location)}-KustoCluster-PrincipalAssignment-${index}'
+  for (principalAssignment, index) in (clusterPrincipalAssignments ?? []): {
```

Comment:
Personally I find `principalAssignments` already does the trick as it's deployed inside a 'cluster'. But there's not spec for one or the other, so it's totally up to you - just couldn't help it 😄 . Feel free to resolve.